docs/porting.md

```diff
@@ -103,10 +103,76 @@ _Important: If you've gotten this far, this is a good time to commit your code b
 
 Pagination is generally unique for almost every API. There's no single method that solves for very different API's approach to pagination.
 
-Most likely you will use `get_next_page_token` to parse and return whatever the "next page" token is for your source, and you'll use `get_url_params` to define how to pass the "next page" token back to the API when asking for subsequent pages.
+Most likely you will use [get_new_paginator](singer_sdk.RESTStream.get_new_paginator) to instantiate a [pagination class](./classes/singer_sdk.pagination.BaseAPIPaginator) for your source, and you'll use `get_url_params` to define how to pass the "next page" token back to the API when asking for subsequent pages.
 
 When you think you have it right, run `poetry run tap-mysource` again, and debug until you are confident the result is including multiple pages back from the API.
 
+You can also add unit tests for your pagination implementation for additional confidence:
+
+```python
+from singer_sdk.pagination import BaseHATEOASPaginator, first
+
+
+class CustomHATEOASPaginator(BaseHATEOASPaginator):
+   def get_next_url(self, response: Response) -> str | None:
```

Comment:
Since you've already created the reference docs pages, I think class docstrings that explain the pattern that each expects would be super helpful. And since many data people won't be familiar with the official pattern names, using simplified explanations and/or references to external explainers will be super helpful, I think.

```suggestion
class CustomHATEOASPaginator(BaseHATEOASPaginator):
    """Paginator for HATEOAS APIs - or "Hypermedia as the Engine of Application State".
    
    This paginator expects responses to have a key "next" with a value
    like "https://api.com/link/to/next-item".
    """"

   def get_next_url(self, response: Response) -> str | None:
```